moped-editor/src/components/GridTable/Filters.js

```diff
@@ -230,215 +129,109 @@ const Filters = ({
-        (filter) => filter.name === field
-      );
-
-      if (!fieldDetails) {
-        filtersNewState[filterId] = generateEmptyField(filterId);
-      } else {
-        // Update field & type
-        filtersNewState[filterId].field = fieldDetails.name;
-        filtersNewState[filterId].type = fieldDetails.type;
-        filtersNewState[filterId].placeholder = fieldDetails.placeholder;
-        filtersNewState[filterId].label = fieldDetails.label;
-
-        // Update Available Operators
-        if (
-          fieldDetails.operators.length === 1 &&
-          fieldDetails.operators[0] === "*"
-        ) {
-          // Add all operators and filter by specific type (defined in fieldDetails.type)
-          filtersNewState[filterId].availableOperators = Object.keys(
-            filtersConfig.operators
-          )
-            .filter(
-              (operator) =>
-                filtersConfig.operators[operator].type === fieldDetails.type
-            )
-            .map((operator) => {
-              return {
-                ...filtersConfig.operators[operator],
-                ...{ id: operator },
-              };
-            });
-        } else {
-          // Append listed operators for that field
-          filtersNewState[filterId].availableOperators =
-            fieldDetails.operators.map((operator) => {
-              return {
-                ...filtersConfig.operators[operator],
-                ...{ id: operator },
-              };
-            });
-        }
-
-        // if the field has a corresponding lookup table, add to filterState
-        if (fieldDetails.lookup) {
-          filtersNewState[filterId].lookup_table =
-            fieldDetails.lookup.table_name;
-          filtersNewState[filterId].lookup_field =
-            fieldDetails.lookup.field_name;
-        }
-        // if it does not, reset the variables to null in case the user is editing an existing filter
-        else {
-          filtersNewState[filterId].lookup_table = null;
-          filtersNewState[filterId].lookup_field = null;
-        }
-      }
+  const handleFilterFieldMenuClick = (filterIndex, field) => {
+    // Clone state
+    const filtersNewState = [...filterParameters];
 
-      // Select the default operator, if not defined select first.
-      if (fieldDetails) {
-        const defaultOperator = getDefaultOperator(fieldDetails);
-        handleFilterOperatorClick(filterId, defaultOperator);
-      }
-      // Update the state
-      setFilterParameters(filtersNewState);
+    // Find the field we need to gather options from
+    const fieldDetails = filtersConfig.fields.find(
+      (filter) => filter.name === field
+    );
+
+    // Update the field and operator values or handle when the clear X icon is clicked
+    if (fieldDetails) {
+      filtersNewState[filterIndex].field = fieldDetails.name;
+      filtersNewState[filterIndex].operator = getDefaultOperator(fieldDetails);
     } else {
-      console.debug(
```

Comment:
imo this debug message isn't needed any longer since the number of filter rows in the UI and the number of items in the `filterParameters` state is equal. There should always be a corresponding filter to a filter row being interacted with.